test/e2e/framework/skip.go

```diff
@@ -108,13 +108,6 @@ func SkipIfMultizone(c clientset.Interface) {
 	}
 }
 
-// SkipUnlessPrometheusMonitoringIsEnabled skips if the prometheus monitoring is not enabled.
-func SkipUnlessPrometheusMonitoringIsEnabled(supportedMonitoring ...string) {
-	if !TestContext.EnablePrometheusMonitoring {
-		skipInternalf(1, "Skipped because prometheus monitoring is not enabled")
-	}
-}
```

Comment:
This PR removes the following from the test context as well:
```go
// Separate Prometheus monitoring deployed in cluster
EnablePrometheusMonitoring bool
```

As far as I know, the e2e test framework is not (should not) be tied to anything in the cluster directory.
I think in general, we should keep this bit.

Reply:
> As far as I know, the e2e test framework is not [...] be tied to anything in the cluster directory.

That's wildly untrue? : - )
The e2e test is tied to EG `log-dump.sh`, and more ...

> As far as I know, the e2e test framework [...] (should not) be tied to anything in the cluster directory.

That's debateable. I don't think we have a definition anywhere of what it should be tied to. It's also tied to lots of cloud providers..

Reply:
(no comment on this particular bool being removed, but I don't think those assertions are accurate)

Reply:
mmm this is cool. So I guess the implementation of some set of features of the e2e test framework may be tied to the way things are configured in the cluster directory.
And maybe this part of the framework needs to be reworked to be more general (?). 

Ah so this is cool. I was wondering if the bits that are being removed would impact the rest of the framework in any way. and going through the code and it seems that the things that are being removed are really only used in the instrumentation tests that are being deleted, so I guess this doesn't affect any other framework functionality.

Made my initial comment thinking that even though the framework may have some bits that assume things were setup using code from the cluster directory, what lives in the cluster directory should not influence the test framework. i.e., we should make things in the framework more general; make it easier for people to consume...
/hold cancel

Reply:
According to https://cs.k8s.io/?q=EnablePrometheusMonitoring&i=nope&files=&repos= EnablePrometheusMonitoring is used at here only and the function SkipUnlessPrometheusMonitoringIsEnabled() is called at `test/e2e/instrumentation/monitoring/prometheus.go` which is removed in this PR as https://cs.k8s.io/?q=SkipUnlessPrometheusMonitoringIsEnabled&i=nope&files=&repos=

So it seems fine to remove this part in this PR for me.